cmp/grid/GridModel.ts

```diff
@@ -1533,6 +1555,24 @@ export class GridModel extends HoistModel {
         }
     }
 
+    private applyExpandToLevel() {
+        if (!this.isReady) return;
+
+        const {agApi} = this;
+        if (this.expandToLevel === 0) {
+            agApi.collapseAll();
+        } else if (this.expandToLevel === this.maxDepth) {
+            agApi.expandAll();
+        } else {
+            agApi.forEachNode(node => {
+                node.setExpanded(node.level < this.expandToLevel);
+            });
```

Comment:
I was wondering how this iteration over every node holds up with very large grids.  Tried our Toolbox Admin > Tests > Grid page.  Allowed it to load its default (if intense) 200k records, the swapped to tree mode.  All OK, but then tried expanding to level three and the tab locked up with 100% CPU for...well, I gave up waiting.

Note that this does not occur with pre-existing expand/collapse all - I can use the toggle in the tree column header and the app is able to execute that operation OK and remain responsive.

Checking first to see how you were looking at / thinking about performance here, happy to discuss what we could do....

Reply:
In similar code in a client app we are setting the `expanded` property on the row nodes directly and then calling `agApi.onGroupExpandedOrCollapsed()` after updating all the nodes, might be worth trying to see if that performs better

Reply:
Will give it a try

Reply:
This appears to improve things but a lot.

Reply:
yes, performance improved, but actually not doing anything